pkg/logs/pipeline/provider.go

```diff
@@ -166,6 +174,12 @@ func (p *provider) NextPipelineChan() chan *message.Message {
 
 // Flush flushes synchronously all the contained pipeline of this provider.
 func (p *provider) Flush(ctx context.Context) {
+	if p.serverless {
+		p.serverlessFlushWg.Add(1)
```

Comment:
At first blush, this logic appears fragile to me, or at least a little underdocumented.

At face value, there could be multiple pipelines in the provider, and each of those pipelines could have multiple endpoints. You're only going to synchronize against a single endpoint the way this is written... which would lead the others to blocking when in serverless mode.

If we can demonstrate that Serverless mode implies there's only ever one pipeline, with a single endpoint... then this is potentially OK, but it should be documented here/somewhere.

Reply:
You were right. I've given this a shot to correct it. Now it's trying to sync a flush to when a payload is sent to all destinations, so it shouldn't matter how many endpoints a user adds now.